src/main/java/org/testeditor/fixture/web/json/BrowserSetupReader.java

```diff
@@ -0,0 +1,147 @@
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+public class BrowserSetupReader {
+    
+    private static final String optionsString = "options";
+    private static final String capabilitiesString = "capabilities"; 
+
+    /**
+     * Reads all available elements. An element is a {@link BrowserSetupElement} like operating system "os" with the 
+     * value "WINDOWS" or "LINUX"s or browser specific "capabilities" or "options". These elements can be used for
+     * setting up a browser before starting a test.
+     * @param fileName of the BrowserElements
+     * @return All Browser or OS specific settings for starting a browser in a test environment.  
+     */
+    public ArrayList<BrowserSetupElement> readElements(String fileName) {
+        FileReader reader = new FileReader();
+        String jsonString = reader.getFileContentAsString(fileName);
+        Gson gson = new GsonBuilder().create();
+        JsonObject obj = gson.fromJson(jsonString, JsonObject.class);
+        ArrayList<BrowserSetupElement> allSetupElements = getAllSetupElements(obj);
+        extractElements(allSetupElements);
+        return allSetupElements;
+    }
+    
+    private void extractElements(ArrayList<BrowserSetupElement> allSetupElements) {
+       
+        for (BrowserSetupElement browserSetupElement : allSetupElements) {
+            List<BrowserSetting> capabilities = new ArrayList<BrowserSetting>();
+            List<BrowserSetting> options = new ArrayList<BrowserSetting>();
+            JsonObject jsonObject = browserSetupElement.getSettingsAsJsonObject();
+            setBrowserSetting(browserSetupElement, options, jsonObject, optionsString);  
+            setBrowserSetting(browserSetupElement, capabilities, jsonObject, capabilitiesString);  
+            setOsName(browserSetupElement, jsonObject);
+            setOsVersion(browserSetupElement, jsonObject);
+            setBrowserName(browserSetupElement, jsonObject);
+            setBrowserVersion(browserSetupElement, jsonObject);
+        }
+    }
+
+    private void setBrowserSetting(BrowserSetupElement browserSetupElement, List<BrowserSetting> settings, 
+            JsonObject jsonObject, String settingType) {
+        JsonElement optionsAsJSonElement = jsonObject.get(settingType);
+        if (optionsAsJSonElement != null) {
+            Set<Map.Entry<String, JsonElement>> settingEntries = optionsAsJSonElement.getAsJsonObject()
+                    .entrySet();
+            for (Map.Entry<String, JsonElement> settingEntry : settingEntries) {
+                BrowserSetting setting = createSettting(settingEntry);
+                if (setting != null) {
+                    settings.add(setting);
+                    if (settingType.equals(capabilitiesString)) {
+                        browserSetupElement.setCapabilities(settings);
```

Comment:
do you really set the capabilities in every loop iteration?
why don't you let the caller do that?
let this method create a list of BrowserSettings and let the caller do the assignment, since the caller knows whether it is a list of capabilities or options, right?